test/pinv.jl

```diff
@@ -0,0 +1,78 @@
+using StaticArrays, Test, LinearAlgebra
+
+tol = 1e-13
+
+@testset "Moore–Penrose inverse (Peseudo-inverse)" begin
+    M1 = @SMatrix [1.5 1.3; 1.2 1.9]
+    N1 = pinv(M1)
+    @test norm(M1*N1*M1 - M1) < tol
+    @test norm(N1*M1*N1 - N1) < tol
+    @test N1 isa SMatrix{2,2,Float64}
+    @test N1 ≈ pinv(Matrix(M1))
+
+    M2 = @SMatrix [1//2 0 0;3//2 5//3 8//7;9//4 -1//3 -8//7;0 0 0]
+    N2 = pinv(M2)
+    @test norm(M2*N2*M2 - M2) < tol
+    @test norm(N2*M2*N2 - N2) < tol
+    @test N2 isa SMatrix{3,4,Float64}
+    @test N2 ≈ pinv(Matrix(M2))
+
+    M3 = SDiagonal(0,1,2,3)
+    N3 = pinv(M3)
+    @test norm(M3*N3*M3 - M3) < tol
+    @test norm(N3*M3*N3 - N3) < tol
+    @test N3 isa Diagonal{Float64, <:SVector{4,Float64}}
+    @test N3 ≈ pinv(Matrix(M3))
+
+    M4 = @SMatrix randn(2,5)
+    N4 = pinv(M4)
+    @test norm(M4*N4*M4 - M4) < tol
+    @test norm(N4*M4*N4 - N4) < tol
+    @test N4 isa SMatrix{5,2,Float64}
+    @test N4 ≈ pinv(Matrix(M4))
+
+    M5 = SMatrix{0,5,Int}()
+    N5 = pinv(M5)
+    @test norm(M5*N5*M5 - M5) < tol
+    @test norm(N5*(M5*N5) - N5) < tol
+    @test N5 isa SMatrix{5,0,Float64}
+    @test N5 ≈ pinv(Matrix(M5))
+
+    M6 = @SMatrix [1/2 0 0;0 5/3 0;0 0 0;0 0 0]
+    N6 = pinv(M6)
+    @test norm(M6*N6*M6 - M6) < tol
+    @test norm(N6*M6*N6 - N6) < tol
+    @test N6 isa SMatrix{3,4,Float64}
+    @test_broken N6 ≈ I(3)/M6
```

Comment:
```suggestion
    @test N6 ≈ I(3)/M6
```

Why did CI pass with this?

Reply:
Thanks for the correction! That was a typo due to my modification of `N6 ≈ pinv(Matrix(M6))`...